lib/ASTGen/Sources/ASTGen/Attrs.swift

```diff
@@ -0,0 +1,1132 @@
+      )
+    }
+  }
+
+  func generateObjCImplementationAttr(attribute node: AttributeSyntax) -> BridgedObjCImplementationAttr? {
+    let name: BridgedIdentifier? = self.generateSingleAttrOption(attribute: node) {
+      self.generateIdentifier($0)
+    }
+    guard let name else {
+      // TODO: Diagnose.
+      return nil
+    }
+
+    return .createParsed(
+      self.ctx,
+      atLoc: self.generateSourceLoc(node.atSign),
+      range: self.generateSourceRange(node),
+      name: name
+    )
+  }
+
+  func generateObjCRuntimeNameAttr(attribute node: AttributeSyntax) -> BridgedObjCRuntimeNameAttr? {
+    let name: BridgedIdentifier? = self.generateSingleAttrOption(attribute: node) {
+      self.generateIdentifier($0)
+    }
+    guard let name else {
+      // TODO: Diagnose.
+      return nil
+    }
+
+    return .createParsed(
+      self.ctx,
+      atLoc: self.generateSourceLoc(node.atSign),
+      range: self.generateSourceRange(node),
+      name: name
+    )
+  }
+
+  func generateOptimizeAttr(attribute node: AttributeSyntax) -> BridgedOptimizeAttr? {
+    let mode: BridgedOptimizationMode? = self.generateSingleAttrOption(
+      attribute: node,
+      {
+        switch $0.rawText {
+        case "speed": return .forSpeed
+        case "size": return .forSize
+        case "none": return .noOptimization
+        default: return nil
+        }
+      }
+    )
+    guard let mode else {
+      return nil
+    }
+    return .createParsed(
+      self.ctx,
+      atLoc: self.generateSourceLoc(node.atSign),
+      range: self.generateSourceRange(node),
+      mode: mode
+    )
+  }
+
+  func generateOriginallyDefinedInAttr(attribute node: AttributeSyntax) -> [BridgedOriginallyDefinedInAttr] {
+    guard
+      // `@_OriginallyDefinedIn` has special argument list syntax.
+      let args = node.arguments?.as(OriginallyDefinedInAttributeArgumentsSyntax.self)
+    else {
+      // TODO: Diagnose.
+      return []
+    }
+
+    fatalError("unimplemented")
+  }
+
+  func generatePrivateImportAttr(attribute node: AttributeSyntax) -> BridgedPrivateImportAttr? {
+    guard
+      // `@_private` has special argument list syntax
+      let args = node.arguments?.as(UnderscorePrivateAttributeArgumentsSyntax.self)
+    else {
+      // TODO: Diagnose
+      return nil
+    }
+
+    guard let fileName = self.generateStringLiteralTextIfNotInterpolated(expr: args.filename) else {
+      // TODO: Diagnose
+      return nil
+    }
+
+    return .createParsed(
+      self.ctx,
+      atLoc: self.generateSourceLoc(node.atSign),
+      attrNameLoc: self.generateSourceLoc(node.attributeName),
+      lParenLoc: self.generateSourceLoc(node.leftParen),
+      fileName: fileName,
+      rParenLoc: self.generateSourceLoc(node.rightParen)
+    )
+  }
+
+  func generateProjectedValuePropertyAttr(attribute node: AttributeSyntax) -> BridgedProjectedValuePropertyAttr? {
+    // `@_dynamicReplacement` has special argument list syntax
```

Comment:
Should this be `@_projectedValueProperty`?

Also, I’m not sure if all the `has special argument list syntax` comments provide any real value.